### DISCUSSION TITLE
A plan to upgrade action protocol to v2

### DISCUSSION THREAD

**Discussion by lyricat** [pandodao/developers.pando.im#9] [Announcements]
We've a plan to upgrade action protocol to v2. 

Draft: https://developers.pando.im/references/action.html

In short, V2 is much clearer and simpler than [current implement](https://github.com/fox-one/4swap-sdk-go/blob/master/mtg/action.go#L113). 

Although the upgrade will be compatible with V1, it is recommended to use V2 in your project. We will provide both Golang and typescript packages for developers.

**Comment by lyricat** [comment]
As mentioned earlier, the Action protocol has been released in the new version of the Pando Web App (4swap). Developers can now try migrating to Action V2.

The new V2 removed encryption and makes the encoding much simpler. Of course, we still provide Go and JS SDKs to assist with the encoding process. It's also not difficult to implement it in other programming languages.

Code example:

- https://developers.pando.im/guide/invoke-actions.html

References:

- https://developers.pando.im/references/action.html
- https://developers.pando.im/references/4swap/action.html

**Comment by coolit** [comment]
I'm neither familiar with Golang nor JS, so hopefully pseudo codes like below can be provided for coders like me: 
checksum = sha256(sha256(header + payload))[:4]
Thanks.
